app/services/SubmissionService.scala

```diff
@@ -53,26 +53,45 @@ class SubmissionService @Inject()(
     }
   }
 
-  def submitPSub(nino: String, taxYears: Seq[TaxYearSelection], claimAmount: Int)
-               (implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Seq[HttpResponse]] = {
+  def submitPSub(nino: String, taxYears: Seq[TaxYearSelection], subscriptions: Map[Int, Seq[PSub]])
+                (implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Seq[HttpResponse]] = {
 
     getTaxYearsToUpdate(nino, taxYears).flatMap {
       claimYears =>
-        futureSequence(claimYears) {
-          taxYearSelection =>
-            val taxYear = TaxYearSelection.getTaxYear(taxYearSelection)
+
+        val psubsToUpdate: Seq[(Int, Seq[PSub])] = claimYears.flatMap {
+          year =>
+            subscriptions.get(getTaxYear(year)).filter(_.nonEmpty).map {
+              psubs =>
+                getTaxYear(year) -> psubs
+            }
+        }
+
```

Comment:
maybe 
`        val psubsToUpdate = for {
          year <- claimYears
          psubs <- subscriptions.get(getTaxYear(year)).filter(_.nonEmpty)
        } yield getTaxYear(year) -> psubs`

Reply:
implemented :)